src/videocapture.cpp

```diff
@@ -578,11 +578,21 @@ shared_ptr<OpaqueCalibration> VideoCapture::get_opaque_calibration() {
 }
 
 bool VideoCapture::get_xu(int ctrl, void* data, int len) {
-  return false;
+  if (!driver_) {
+    return false;
+  }
+  
+  auto res = driver_->get_prop(ctrl, (double*) data);
```

Comment:
I get the feeling this was answered by your comment which has now been removed... I assume `get_prop` actually writes different types of data but it gets passed in as a `double*` for reasons?

I'm assuming `get_prop` here actually writes a `uint32_t` to that address (hence the type shuffling in `driver_rigel`)?

Reply:
Apologies, I had completely the wrong mental model for how these parts fit together. The code makes sense as-is.